src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SecurityChannelFactory.cs

```diff
@@ -0,0 +1,845 @@
+            }
+
+            public override T GetProperty<T>()
+            {
+                if (typeof(T) == typeof(ChannelParameterCollection))
+                {
+                    return (T)(object)_channelParameters;
+                }
+
+                return base.GetProperty<T>();
+            }
+        }
+
+        class SecurityOutputChannel : ClientSecurityChannel<IOutputChannel>, IOutputChannel
+        {
+            public SecurityOutputChannel(ChannelManagerBase factory, SecurityProtocolFactory securityProtocolFactory, IOutputChannel innerChannel, EndpointAddress to, Uri via)
+                : base(factory, securityProtocolFactory, innerChannel, to, via)
+            {
+            }
+
+            public IAsyncResult BeginSend(Message message, AsyncCallback callback, object state)
+            {
+                return BeginSend(message, DefaultSendTimeout, callback, state);
+            }
+
+            public IAsyncResult BeginSend(Message message, TimeSpan timeout, AsyncCallback callback, object state)
+            {
+                throw ExceptionHelper.PlatformNotSupported();   // Issue #31 in progress
+                //ThrowIfFaulted();
+                //ThrowIfDisposedOrNotOpen(message);
+                //return new OutputChannelSendAsyncResult(message, SecurityProtocol, InnerChannel, timeout, callback, state);
+            }
+
+            public void EndSend(IAsyncResult result)
+            {
+                throw ExceptionHelper.PlatformNotSupported();   // Issue #31 in progress
+                // OutputChannelSendAsyncResult.End(result);
+            }
+
+            public void Send(Message message)
+            {
+                Send(message, DefaultSendTimeout);
+            }
+
+            public void Send(Message message, TimeSpan timeout)
+            {
+                ThrowIfFaulted();
+                ThrowIfDisposedOrNotOpen(message);
+                TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+                SecurityProtocol.SecureOutgoingMessage(ref message, timeoutHelper.RemainingTime());
+                InnerChannel.Send(message, timeoutHelper.RemainingTime());
+            }
+        }
+
+        sealed class SecurityOutputSessionChannel : SecurityOutputChannel, IOutputSessionChannel
+        {
+            public SecurityOutputSessionChannel(ChannelManagerBase factory, SecurityProtocolFactory securityProtocolFactory, IOutputSessionChannel innerChannel, EndpointAddress to, Uri via)
+                : base(factory, securityProtocolFactory, innerChannel, to, via)
+            {
+            }
+
+            public IOutputSession Session
+            {
+                get
+                {
+                    return ((IOutputSessionChannel)InnerChannel).Session;
+                }
+            }
+        }
+
+        class SecurityRequestChannel : ClientSecurityChannel<IRequestChannel>, IRequestChannel
```

Comment:
Yes, I'll make that change and stub them for now.

---
In reply to: [78485068](https://github.com/dotnet/wcf/pull/1515#discussion_r78485068) [](ancestors = 78485068)